test/server/cards/03_TWI/leaders/AsajjVentressUnparalleledAdversary.spec.ts

```diff
@@ -0,0 +1,137 @@
+                        resources: 3,
+                    },
+                    player2: {
+                        groundArena: ['admiral-yularen#advising-caution'],
+                    },
+                });
+            });
+
+            it('should initiate attack but does not give +1/+0 as we do not play any event this phase', function () {
+                const { context } = contextRef;
+
+                // initiate attack
+                context.player1.clickCard(context.asajjVentress);
+                expect(context.player1).toBeAbleToSelectExactly([context.battlefieldMarine, context.greenSquadronAwing]);
+                context.player1.clickCard(context.battlefieldMarine);
+                context.player1.clickCard(context.p2Base);
+
+                expect(context.player2).toBeActivePlayer();
+                expect(context.asajjVentress.exhausted).toBeTrue();
+                expect(context.p2Base.damage).toBe(3);
+            });
+
+            it('should initiate attack and give +1/+0 as we play an event this phase', function () {
+                const { context } = contextRef;
+
+                // play an event
+                context.player1.clickCard(context.smugglersAid);
+                context.player2.passAction();
+
+                // initiate attack
+                context.player1.clickCard(context.asajjVentress);
+                expect(context.player1).toBeAbleToSelectExactly([context.battlefieldMarine, context.greenSquadronAwing]);
+                context.player1.clickCard(context.battlefieldMarine);
+                context.player1.clickCard(context.p2Base);
+
+                expect(context.player2).toBeActivePlayer();
+                expect(context.asajjVentress.exhausted).toBeTrue();
+                expect(context.p2Base.damage).toBe(4);
+            });
+        });
+
+        describe('Asajj Ventress\'s leader deployed ability', function () {
+            beforeEach(function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        leader: { card: 'asajj-ventress#unparalleled-adversary', deployed: true },
+                    },
+                    player2: {
+                        groundArena: ['battlefield-marine'],
+                    },
+                });
+            });
+
+            it('should not have +1/+0 and deals before defender as we do not play any event this phase', function () {
+                const { context } = contextRef;
+
+                context.player1.clickCard(context.asajjVentress);
+                context.player1.clickCard(context.battlefieldMarine);
+
+                expect(context.player2).toBeActivePlayer();
+                expect(context.battlefieldMarine).toBeInZone('discard');
+                expect(context.asajjVentress.damage).toBe(3);
+            });
+        });
+
+        describe('Asajj Ventress\'s leader deployed ability', function () {
```

Comment:
We can combine this and the next describe into one and just have all three integration tests in the first describe.